cpp/src/slice2js/Gen.cpp

```diff
@@ -2857,33 +2883,69 @@ Slice::Gen::TypeScriptVisitor::visitStructStart(const StructPtr& p)
     _out << nl << "constructor" << spar;
     for (const auto& dataMember : dataMembers)
     {
-        _out << (fixDataMemberName(dataMember->name(), false, false) + "?:" + typeToTsString(dataMember->type()));
+        // TODO why are all parameters optional?
```

Comment:
What is the implementation of this constructor?

Reply:
This is the generated constructor for struct types. I will review why it is that way.